app.js

```diff
@@ -12,9 +12,19 @@ const data = {
   vegetables: []
 }
 
+// Vegetables
+
 app.get('/vegetables', (req, res, next) => {
   const { vegetables } = data
-  res.json(vegetables)
+  const { name } = req.query
+  const vegetable = vegetables.filter(veggie => veggie.name.includes(name))
+
+  if (!name) {
+    const message = `Could not find vegetable with name of ${name}`
+    next({ status: 404, message })
```

Comment:
We actually don't want to stop the request here. If there's no name, that's ok, we should return all of the vegetables.